torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1631,3 +1643,19 @@ def test_function_not_on_callee(self):
                 AttributeError, "RPC pickler does not serialize"
             ):
                 rpc.rpc_sync(callee_worker, foo_add, args=())
+
+    @dist_init
+    def test_async_script_udf(self):
+        for i in range(1, 3):
+            future = rpc.rpc_async(
+                "worker{}".format((self.rank + 1) % self.world_size),
```

Comment:
I guess there is no easy way for us to test if this is indeed launched asynchronously on the server side?

Reply:
Well, I added printfs to convince myself that this was happening, and that we were hitting all the codepaths. But...

If you have ideas for what to throw into python here (particularly that will also work in TorchScript), that would be good, but otherwise, it feels tricky.